packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -1162,31 +1161,22 @@ export class ObjectsInRedisClient {
         );
     }
 
-    async _readFile(
-        id: string,
-        name: string,
-        options: CallOptions,
-        callback: ioBroker.ReadFileCallback,
-        meta: MetaObject
-    ) {
+    async _readFile(id: string, name: string, options: CallOptions, meta: MetaObject): ioBroker.ReadFilePromise {
```

Comment:
Does TS not complain that the return type of an async function must literally be `Promise<...>` and not a type alias? Or did they make this check smarter?

Reply:
TS compiles successfully an does not complain, the type is refering to a Promise 

```
        type ReadFilePromise = Promise<{ file: string | Buffer; mimeType: string }>;
```

but you cannot do `ReadFilePromise | Promise<void>` then TS complains that you have to return the global Promise type. so you would have to get the args and make `Promise<ReadFilePromiseArgs|void>`